controllers/terraform_controller.go

```diff
@@ -2220,6 +2220,37 @@ func (r *TerraformReconciler) runnerPodSpec(terraform infrav1.Terraform, tlsSecr
 	vTrue := true
 	vUser := int64(65532)
 
+	podVolumes := []corev1.Volume{
+		{
+			Name: "temp",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+		{
+			Name: "home",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+	if len(terraform.Spec.RunnerPodTemplate.Spec.Volumes) != 0 {
+		podVolumes = append(podVolumes, terraform.Spec.RunnerPodTemplate.Spec.Volumes...)
+	}
+	podVolumeMounts := []corev1.VolumeMount{
+		{
+			Name:      "temp",
+			MountPath: "/tmp",
+		},
+		{
+			Name:      "home",
+			MountPath: "/home/runner",
+		},
+	}
+	if len(terraform.Spec.RunnerPodTemplate.Spec.VolumeMounts) != 0 {
+		podVolumeMounts = append(podVolumeMounts, terraform.Spec.RunnerPodTemplate.Spec.VolumeMounts...)
+	}
```

Comment:
We should not allow users to override the defaults (/tmp and /home) because it's going to break the current Terraform behaviour.

- /tmp is being used by the Runner to init the workspace
- /home/runner is required to be writable by some Terraform providers